src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -126,9 +146,15 @@ public static object GetObjectForIUnknown(IntPtr pUnk)
         }
 
         [SupportedOSPlatform("windows")]
-        public static object? GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
+        public static unsafe object? GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
         {
-            throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+            if (pSrcNativeVariant == IntPtr.Zero)
+            {
+                return null;
+            }
+
+            Variant* data = (Variant*)pSrcNativeVariant;
+            return data->ToObject();
```

Comment:
This is going to call `GetObjectForNativeVariant` in some cases that will lead to infinite recursion.

Reply:
I am not sure why `ToObject` handles some of the cases inline and does not just call `GetObjectForNativeVariant` all the time.

Reply:
Upon more investigation I would think I incorrectly use `Variant.CopyFromIndirect`. Only place where I see it's usage is in https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Common/src/System/Runtime/InteropServices/ComEventsSink.cs#L231 
I thought I do smart, move, apparently I have to reimplement `GetObjectForNativeVariant` in similar manner.

Also I would like to ask @jkotas what should I do with test failing due to reflection. I assume reflection needed for type checks `is typeof(byte)`. I do not see a way how to solve that, maybe move test to separate class where reflection would not be blocked? Please advice.

Reply:
`is typeof(byte)` does not use reflection. 

The tests failures are caused by invalid IL produced by the new marshaller that you have added.